packages/components/src/higher-order/with-notices/types.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Internal dependencies
+ */
+import type { NoticeListProps } from '../../notice/types';
+
+export type WithNoticeProps = {
+	noticeList: NoticeListProps[ 'notices' ];
```

Comment:
We may want to update the README, since there's no mention that the `noticeList` property is passed to the `OriginalComponent` alongside the `noticeOperations` and `noticeUI` props.